src/commonTest/kotlin/it/krzeminski/snakeyaml/engine/kmp/CommonTestResourcesAccess.kt

```diff
@@ -12,24 +22,87 @@ import okio.ByteString
+        listOrNull(dir) ?: throw IOException("Cannot list $dir")
+
+    override fun listOrNull(dir: Path): List<Path>? {
+        val node = traverseResourcesMap(dir) ?: return null
+        if (node !is Map<*, *>) {
+            return null
+        }
+        @Suppress("UNCHECKED_CAST")
+        return (node as Map<String, Any?>).keys.map { dir.resolve(it) }
+    }
+
+    override fun metadataOrNull(path: Path): FileMetadata? =
+        traverseResourcesMap(path)?.let { node ->
+            FileMetadata(
+                isDirectory = node is Map<*, *>,
+                isRegularFile = node !is Map<*, *>,
+            )
+        }
+
+    override fun source(file: Path): Source =
+        traverseResourcesMap(file)?.let {
+            (Buffer().write(it as ByteString) as Source)
+        } ?: throw IOException("Cannot read $file")
+
+    override fun canonicalize(path: Path): Path =
+        throw NotImplementedError("This operation is not supported by this simple implementation of the file system.")
+
+    override fun openReadOnly(file: Path): FileHandle =
+        throw NotImplementedError("This operation is not supported by this simple implementation of the file system.")
+
+    override fun openReadWrite(file: Path, mustCreate: Boolean, mustExist: Boolean): FileHandle =
+        throw NotImplementedError("This operation is not supported by this simple implementation of the file system.")
+
+    override fun sink(file: Path, mustCreate: Boolean): Sink =
+        throw NotImplementedError("This operation is not supported by this simple implementation of the file system.")
+
+    override fun appendingSink(file: Path, mustExist: Boolean): Sink =
+        throw NotImplementedError("This operation is not supported by this simple implementation of the file system.")
+
+    override fun createDirectory(dir: Path, mustCreate: Boolean) =
+        throw NotImplementedError("This operation is not supported by this simple implementation of the file system.")
+
+    override fun atomicMove(source: Path, target: Path) =
+        throw NotImplementedError("This operation is not supported by this simple implementation of the file system.")
+
+    override fun delete(path: Path, mustExist: Boolean) =
+        throw NotImplementedError("This operation is not supported by this simple implementation of the file system.")
+
+    override fun createSymlink(source: Path, target: Path) =
+        throw NotImplementedError("This operation is not supported by this simple implementation of the file system.")
+}
+
+/**
+ * Returns a map or a leaf (file) in the resources map, pointed to by [path],
+ * or `null` if the file or directory cannot be found.
+ */
+private fun traverseResourcesMap(path: Path): Any? {
+    var map: Any? = CommonTestResources.resourcesMap
+    for (segment in path.segments) {
+        if (map is Map<*, *>) {
+            if (segment in map) {
+                map = map[segment]
+            } else {
+                return null
+            }
```

Comment:
suggestion: not sure if it is more readable but will interact with the map only once

```kotlin
map = map[segment] ?: return null
```

Reply:
https://github.com/krzema12/snakeyaml-engine-kmp/pull/365/commits/ebef078ce0acdf79d85ad8d03bede3b7c6d813c5